### DISCUSSION TITLE
Routes does not work on zeit server while do work in dev

### DISCUSSION THREAD

**Discussion by svobik7** [vercel/vercel#4797] [Help]
We have been using vercel for a while now. We had been using it even before dynamic routing was introduced. So therefore we have custom routes defines in vercel.json (now.json) file. Everything works fine since today's (7.7.2020) first deployment.

We have got Route A:

```js
{
      "src": "/(cs|en|es|sk)/admin/(.*)\\.htm",
      "dest": "/$1/admin/$2"
},
```

Which is pointing to page with `getInitialProps`. Since today I am getting 404 on zeit server while on dev env everything works.

Then we have got Route B:

```js
{
      "src": "/(cs|en|es|sk)/(.*)(a|b|c|d|e|f)([0-9]{1,2})\\.htm",
      "dest": "/$1/basic/$3$4"
},
```

Which is pointing to static page (page without getInitialProps or any other special functions). This still works as expected.

Using: `Vercel CLI 19.1.1`

Is there any change in zeit servers so that routes not work in our case?

**UPDATE**:
Page with `getInitialProps` which is targeted by custom route does not work during SSR only. It works during client side transition.

**Comment by paulogdm** [comment]
Hello,

Please reach out to https://vercel.com/support/request so we can exchange ideas and optimize your configuration. I believe we can kill your "now.json" in favor of Zero Config.

**Comment by svobik7** [comment]
@paulogdm thank you. 

In the meantime I found funny thing. Deploying using command `vercel --local-config vercel.prod.json --prod` did not work properly on vercel server BUT when I deploy with command `vc --prod` and have just `vercel.json` file in my project root everything works well even on vercel server.

I also found that vercel routing is now considered legacy (https://github.com/vercel/vercel/blob/master/errors/next-legacy-routes-optimized-lambdas.md). So that I am working on dropping routes completely right now. As we have multilangual site it will take some time to drop it. 

If there is any problem I will create support request :-)
